transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.plugin.tasks;
+
+import com.github.jengelman.gradle.plugins.shadow.relocation.RelocateClassContext;
+import com.github.jengelman.gradle.plugins.shadow.relocation.RelocatePathContext;
+import com.github.jengelman.gradle.plugins.shadow.relocation.SimpleRelocator;
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+
+
+/**
+ * This code is borrowed from the Shade plugin which has not been open-sourced yet. Once the plugin is open-sourced, we
```

Comment:
Remove and provide proper java doc.

Reply:
I'm negative to remove but I'm ok with a separate github issue created

Reply:
Added java doc